src/Cli/dotnet/commands/dotnet-add/dotnet-add-package/Program.cs

```diff
@@ -41,7 +41,7 @@ public override int Execute()
                 try
                 {
                     // Create a Dependency Graph file for the project
-                    tempDgFilePath = Path.GetTempFileName();
+                    tempDgFilePath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
```

Comment:
The referenced link is about syscall and C library function, which doesn't apply to .NET BCL methods.
The new code is much less secure, since `GetTempFileName` would do some attempt to check the file name is unique, while `GetRandomFileName` is totally disconnected from the path and no check is applied.

Reply:
@huoyaoyuan even if the specific APIs are not being used directly, the description of risks still might apply, and in fact I strongly believe it does. If you read the behaviour of the denial of service scenarios it described, it applies equally to .NET's `Path.GetTempFileName()` method, as it works the same as the `mkstemp()` API mentioned (which runtime's `Path.Unix.cs` appears to use in its implementation of `Path.GetTempFileName()`).

However, this PR does not address that. `Path.GetRandomFileName()` has a max bound on how many unique filenames it might generate. So, if an attacker pre-creates every possible filename, there would be a denial of service attack if this PR actually validated that the file doesn't exist on creation. Perhaps `GetRandomFileName()` has a larger number of filenames it generates than `GetTempFileName()` does, I don't know, so maybe it makes the DoS potential less likely.

But, this PR doesn't validate if the file exists, therefore opening it up to the risks that the OWASP article itself describes as more serious. Although in this specific usage, the file gets overwritten and file length truncated, before it ever gets read, so it's not actually vulnerable to what the OWASP article describes. So, a more realistic risk is that `dotnet add package` overwrites another program's temp file, or crashes because it's unable to overwrite the file.

Therefore, while this PR might be well intentioned, it's likely to increase risks, rather than mitigate any since, as @KalleOlaviNiemitalo wrote in another comment, it does not create the file as the OWASP article describes. It's an example of why trusting automated reporting from tools needs to be validated, rather than trusted as always correct, and the fix needs to be implemented carefully.